### PR TITLE
Add SDK JS CLI usage and WSL for windows

### PR DESCRIPTION
**Pull request by idea404**
Addresses: #1326

### CODE REVIEW

File: docs/sdk/js/cli/cli.md [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+
+**Commands**
+
+| Command                                   | Description                                                                                                                    |
+| ----------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------ |
+| [`near-sdk-js build`](#build)               | stores a full access key locally using [NEAR Wallet](https://wallet.testnet.near.org/) |
+| [`near-sdk-js validateContract`](#validate-contract)                 | displays all access keys and their details for a given account |
+| [`near-sdk-js checkTypescript`](#check-ts) | generates a local key pair **or** shows public key & [implicit account](/concepts/basics/accounts/account-id#implicit-accounts) |
+| [`near-sdk-js createJsFileWithRollup`](#create-js-file)           | adds a new access key to an account |
+| [`near-sdk-js transpileJsAndBuildWasm`](#transpile-js-to-wasm)     | deletes an access key from an account |
```
Comment (bucanero) [inline docs/sdk/js/cli/cli.md:25]: Maybe I'm mistaken, but the descriptions under the "Description" column seem out of place with the actual commands on the left. (Example: "transpile js" >> "deletes an access key from an account")

File: docs/sdk/js/cli/cli.md [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+
+## Setup {#setup}
+
+### Installation {#installation}
+
+> Make sure you have a current version of `npm` and `NodeJS` installed.
+
+#### Mac and Linux {#mac-and-linux}
+
+1. Install `npm` and `node` using a package manager like `nvm` as sometimes there are issues using Ledger due to how OS X handles node packages related to USB devices. [[click here]](https://nodejs.org/en/download/package-manager/)
```
Comment (bucanero) [inline docs/sdk/js/cli/cli.md:37]:
```suggestion
1. Install `npm` and `node` using a [package manager](https://nodejs.org/en/download/package-manager/) like `nvm` as sometimes there are issues using Ledger due to how macOS handles node packages related to USB devices.
```

File: docs/sdk/js/cli/cli.md [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+<strong>heads up</strong><br /><br />
+
+Copy/pasting can be a bit odd using `WSL`.
+
+- "Quick Edit Mode" will allow right-click pasting.
+- Depending on your version there may be another checkbox allowing `Ctrl` + `V` pasting as well.
+
+![Windows option called Quick Edit allow right-click pasting in WSL](/docs/assets/windows-quickedit-mode.png)
+
+</blockquote>
```
Comment (bucanero) [inline docs/sdk/js/cli/cli.md:70]:
```suggestion
:::info heads up

Copy/pasting can be a bit odd using `WSL`.

- "Quick Edit Mode" will allow right-click pasting.
- Depending on your version there may be another checkbox allowing `Ctrl` + `V` pasting as well.

![Windows option called Quick Edit allow right-click pasting in WSL](/docs/assets/windows-quickedit-mode.png)

:::
```

File: docs/sdk/js/cli/cli.md [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+
+---
+
+## Commands {#commands}
+
+### `near-sdk-js build` {#build}
+
+Build a NEAR JS Smart-contract, specifying the source, target, package.json, and tsconfig.json files. If none are specified, the default values are used. The argument default values are:
```
Comment (bucanero) [inline docs/sdk/js/cli/cli.md:78]:
```suggestion
Build a NEAR JS Smart-contract, specifying the source, target, `package.json`, and `tsconfig.json` files. If none are specified, the default values are used. The argument default values are:
```

File: docs/sdk/js/cli/cli.md [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+
+**Example:**
+
+```bash
+near-sdk-js build src/main.ts out/main.wasm package.json tsconfig.json --verbose true --generateABI true
+```
+
+### `near-sdk-js validateContract` {#validate-contract}
+
+Validate a NEAR JS Smart-contract. Validates the contract by checking that all parameters are initialized in the constructor. Works only for typescript.
```
Comment (bucanero) [inline docs/sdk/js/cli/cli.md:98]:
```suggestion
Validate a NEAR JS Smart-contract. Validates the contract by checking that all parameters are initialized in the constructor. Works only for TypeScript.
```

File: docs/sdk/js/cli/cli.md [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+```
+
+</p>
+</details>
+
+---
+
+### `near-sdk-js createJsFileWithRollup` {#create-js-file}
+
+Create intermediate javascript file for later processing with QJSC.
```
Comment (bucanero) [inline docs/sdk/js/cli/cli.md:147]:
```suggestion
Create an intermediate JavaScript file for later processing with QJSC.
```

File: docs/sdk/js/cli/cli.md [hunk 7]
```diff
@@ -0,0 +1,191 @@
...
+
+npx near-sdk-js createJsFileWithRollup src/index.ts
+[createJsFileWithRollup] › …  awaiting  Creating src/index.ts file with Rollup...
+
+</p>
+</details>
+
+### `near-sdk-js transpileJsAndBuildWasm` {#transpile-js-to-wasm}
+
+Create intermediate javascript file for later processing with QJSC
```
Comment (bucanero) [inline docs/sdk/js/cli/cli.md:170]:
```suggestion
Create an intermediate JavaScript file for later processing with QJSC
```

File: docs/sdk/js/cli/cli.md [hunk 8]
```diff
@@ -0,0 +1,191 @@
...
+```
+npx near-sdk-js validateContract src/index.ts
+[validate] › …  awaiting  Validating src/index.ts contract...
+```
+
+---
+
+### `near-sdk-js checkTypescript` {#check-ts}
+
+Run TSC with some cli flags - warning - ignores tsconfig.json.
```
Comment (bucanero) [inline docs/sdk/js/cli/cli.md:120]:
```suggestion
Run TSC with some CLI flags.

 :::note warning
 This command ignores `tsconfig.json`.
 :::
```

File: docs/sdk/js/cli/cli.md [hunk 9]
```diff
@@ -0,0 +1,191 @@
...
+near-sdk-js checkTypescript src/main.ts --verbose true
+```
+
+<details>
+<summary>**Example Response:**</summary>
+<p>
+
+```bash
+npx near-sdk-js checkTypescript src/index.ts
+[checkTypescript] › …  awaiting  Typechecking src/index.ts with tsc...
```
Comment (bucanero) [inline docs/sdk/js/cli/cli.md:137]:
```suggestion
**Example Response:**

```bash
npx near-sdk-js checkTypescript src/index.ts
[checkTypescript] › …  awaiting  Typechecking src/index.ts with tsc...
```

File: docs/sdk/js/cli/cli.md [hunk 10]
```diff
@@ -0,0 +1,191 @@
...
+
+**Example:**
+
+```bash
+near-sdk-js transpileJsAndBuildWasm src/main.js out/main.wasm --verbose true
+```
+
+<details>
+<summary>**Example Response:**</summary>
+<p>
```
Comment (bucanero) [inline docs/sdk/js/cli/cli.md:183]:
```suggestion
**Example Response:**
```

File: docs/sdk/js/cli/cli.md [hunk 11]
```diff
@@ -0,0 +1,191 @@
...
+
+```bash
+npx near-sdk-js transpileJsAndBuildWasm
+[transpileJsAndBuildWasm] › ✔  success   Generated build/contract.wasm contract successfully!
+```
+
+</p>
+</details>
```
Comment (bucanero) [inline docs/sdk/js/cli/cli.md:191]:
```suggestion
```

File: docs/sdk/js/intro.md [hunk 12]
```diff
@@ -5,6 +5,22 @@ title: "Getting Started"
 
 # Getting Started
 
+:::info Using the JS SDK on Windows
+You can develop smart contracts on Windows using Windows Subsystem for Linux (WSL2).
```
Comment (bucanero) [inline docs/sdk/js/intro.md:9]:
```suggestion
:::tip Using the JS SDK on Windows
You can develop smart contracts on Windows using Windows Subsystem for Linux (WSL2).
:::
```

File: docs/sdk/js/cli/cli.md (hunk 1 above)
Comment (idea404) [reply docs/sdk/js/cli/cli.md:25]: Ah well spotted! Let me fix that.

**Comment by idea404** [comment]
Thanks for the comments @bucanero ! Fixes are in

**Review by bucanero — APPROVED** [review]

**Pull request merged**